### PR TITLE
feat(metrics-extraction): Add support for multiple-aggregate timeseries

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
Noticed while spot-checking that we still need to fill in multiple query support for aggregates __without__ group-by (which is top-events and a specialized query builder). This:
- Adds multiple aggregate support (by iterating over specs instead of just picking the first one like before)
- Adds both a metrics builder test and a more integrated test to the endpoint for better coverage

#### Other
- Depends on https://github.com/getsentry/sentry/pull/60176 since I switched the test class to just be for widget on-demand support.
- Moved `_metric_layer_result` method out of the "TopEvents" QueryBuilder and into it's parent (regular Timeseries), since the only main difference is whether it pulls out groups or not, which is now a bool (see comment)

### CODE REVIEW

File: src/sentry/search/events/builder/metrics.py [hunk 1]
```diff
@@ -1444,51 +1444,41 @@ def run_query(self, referrer: str, use_cache: bool = False) -> Any:
             )
 
             try:
+                metrics_queries = []
```
Comment (k-fish) [inline src/sentry/search/events/builder/metrics.py:1447]: This `run_query` branch is mostly the same as what we did for top events, except the queries being appended to the queries array is obviously different.

File: src/sentry/search/events/builder/metrics.py [hunk 2]
```diff
@@ -1510,6 +1500,60 @@ def run_query(self, referrer: str, use_cache: bool = False) -> Any:
...
+                if meta["name"] not in seen_metrics_metas:
+                    seen_metrics_metas[meta["name"]] = True
+                    metric_layer_result["meta"].append(meta)
+            for meta in metric_layer_result["meta"]:
+                if meta["name"] == "bucketed_time":
+                    meta["name"] = "time"
+
+            # Top timeseries uses all groups, but regular timeseries only need the first group.
+            _groups = metrics_data.get("groups", [{}])
+            groups = _groups[:1] if use_first_group_only else _groups
```
Comment (k-fish) [inline src/sentry/search/events/builder/metrics.py:1527]: The only difference converting metric layer result between TopTimeseries and regular Timeseries

File: tests/sentry/search/events/builder/test_metrics.py [hunk 3]
```diff
@@ -2724,14 +2724,63 @@ def test_on_demand_map_with_multiple_selected(self):
...
+        )
+        spec_two = OnDemandMetricSpec(
+            field=field_two, groupbys=groupbys, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY
+        )
+
+        for day in range(0, 4):
+            self.store_on_demand_metric(
+                day * 250,
+                spec=spec,
+                timestamp=self.start + datetime.timedelta(days=0),
```
Comment (k-fish) [inline tests/sentry/search/events/builder/test_metrics.py:2752]: Set day to = 0 here, since for distribution we have to make more than one metric for it to work properly for a time bucket

File: tests/snuba/api/endpoints/test_organization_events_stats_mep.py [hunk 4]
```diff
@@ -1150,3 +1149,67 @@ def test_top_events_with_transaction_on_demand_and_no_environment(self):
...
+        indexer.record(UseCaseID.TRANSACTIONS, self.organization.id, "transaction.duration")
+
+        for count in range(0, 4):
+            self.store_on_demand_metric(
+                count * 100,
+                spec=spec,
+                timestamp=self.day_ago + timedelta(hours=1),
+            )
+            self.store_on_demand_metric(
+                count * 200.0,
```
Comment (k-fish) [inline tests/snuba/api/endpoints/test_organization_events_stats_mep.py:1184]: Same as above, holding `hours=1`, these will store [0,200,400,..] in the single hour time bucket, which p75 will approximate.

File: src/sentry/search/events/builder/metrics.py [hunk 5]
```diff
@@ -1510,6 +1500,60 @@ def run_query(self, referrer: str, use_cache: bool = False) -> Any:
             "meta": [{"name": key, "type": value} for key, value in meta_dict.items()],
         }
 
+    def _metric_layer_result(
+        self, metrics_data_list: Sequence[dict[str, Any]], use_first_group_only: bool
+    ) -> Any:
+        """The metric_layer returns results in a non-standard format, this function changes it back to the expected
```
Comment (armenzg) [inline src/sentry/search/events/builder/metrics.py:1506]: Thank you for the docstring! 💃🏻

File: src/sentry/search/events/builder/metrics.py (hunk 2 above)
Comment (armenzg) [reply src/sentry/search/events/builder/metrics.py:1527]: Good refactor 👍🏻

File: tests/sentry/search/events/builder/test_metrics.py [hunk 6]
```diff
@@ -2724,14 +2724,61 @@ def test_on_demand_map_with_multiple_selected(self):
             query=query_str,
             selected_columns=["eps()", "epm()", "not_on_demand"],
             config=QueryBuilderConfig(
-                on_demand_metrics_enabled=True, on_demand_metrics_type=MetricSpecType.SIMPLE_QUERY
+                on_demand_metrics_enabled=True, on_demand_metrics_type=MetricSpecType.DYNAMIC_QUERY
```
Comment (armenzg) [inline tests/sentry/search/events/builder/test_metrics.py:2727]: Why this change?

File: tests/snuba/api/endpoints/test_organization_events_stats_mep.py [hunk 7]
```diff
@@ -1150,3 +1149,64 @@ def test_top_events_with_transaction_on_demand_and_no_environment(self):
...
+            spec._query_str_for_hash
+            == "event.measurements.fcp.value;{'name': 'event.duration', 'op': 'gte', 'value': 100.0}"
+        )
+        assert (
+            spec_two._query_str_for_hash
+            == "event.measurements.lcp.value;{'name': 'event.duration', 'op': 'gte', 'value': 100.0}"
+        )
+
+        # Include transaction duration in indexer since it's in the query.
+        indexer.record(UseCaseID.TRANSACTIONS, self.organization.id, "transaction.duration")
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_organization_events_stats_mep.py:1172]: Why do we not need it in some other test cases? In other words, why do we need it in this test case?

**Review by armenzg — COMMENTED** [review]
It looks good. I can approve now or wait until the test is fixed. Whatever you prefer me to do.

File: tests/sentry/search/events/builder/test_metrics.py (hunk 6 above)
Comment (k-fish) [reply tests/sentry/search/events/builder/test_metrics.py:2727]: We're the only ones who use multiple aggregates (alerts is only 1 aggregate I believe), these multiple selected will only ever be called with dynamic query, so it's more appropriate for this to be `DYNAMIC_QUERY`

**Review by armenzg — APPROVED** [review]
